python/nx-cugraph/nx_cugraph/algorithms/centrality/eigenvector.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import networkx as nx
+import numpy as np
+import pylibcugraph as plc
+
+from nx_cugraph.convert import _to_graph
+from nx_cugraph.utils import (
+    _dtype_param,
+    _get_float_dtype,
+    networkx_algorithm,
+    not_implemented_for,
+)
+
+__all__ = ["eigenvector_centrality"]
+
+
+@not_implemented_for("multigraph")
+@networkx_algorithm(extra_params=_dtype_param)
+def eigenvector_centrality(
+    G, max_iter=100, tol=1.0e-6, nstart=None, weight=None, *, dtype=None
+):
+    """`nstart` parameter is not used."""
+    G = _to_graph(G, weight, np.float32)
+    if len(G) == 0:
+        raise nx.NetworkXPointlessConcept(
+            "cannot compute centrality for the null graph"
+        )
+    if dtype is not None:
+        dtype = _get_float_dtype(dtype)
+    elif weight in G.edge_values:
+        dtype = _get_float_dtype(G.edge_values[weight].dtype)
+    else:
+        dtype = np.float32
+    if nstart is not None:
+        # Check if given nstart is valid even though we don't use it
+        nstart = G._dict_to_nodearray(nstart, dtype=dtype)
+        if (nstart == 0).all():
+            raise nx.NetworkXError("initial vector cannot have all zero values")
+        if (total := nstart.sum()) == 0:
+            raise ZeroDivisionError
+        nstart /= total
```

Comment:
I'm not following why we're processing `nstart` like this. As a user, this makes me think it's being used if I see these exceptions. 

Why isn't this raising `NotImplementedError`, or even a warning that it's not being used if set?

Reply:
Oh, right, I guess I don't need to normalize `nstart`. I can comment this line out (but not delete it), since it's doing unnecessary work.

We handle `nstart` as if we are going to use it to prepare for when we will actually use it. If we ignore it completely, then users can write code that works today, but will fail for newer versions or when running with a different backend. Simply put: it's better to raise when raising is the correct behavior. I don't think we should raise `NotImplementedError`, b/c then  e.g. networkx will raise, so why convert to another backend or nx if we know it's going to raise an error?

I expect operating on node vectors like this to be super fast. This check is cheap.

Reply:
Right, I'm not worried about if the check is expensive as much as I am about meeting expectations. I see you updated the docstring on this arg so IMO that's sufficient for now.

> then users can write code that works today, but will fail for newer versions or when running with a different backend

I think this is the real problem we'll have to deal with: how do we support drop-in compatibility when NX and different backends have varying levels of support for different args?  I suspect the approach we're taking here won't scale (e.g. if NX adds an arg or changes what it considers valid for an arg, will backends be expected to track these changes and update their code?), but that's outside the scope of this PR so we may have to have a bigger discussion later. cc @mriduls